### PR TITLE
feat(site): search docs 

### PR DESCRIPTION
**Pull request by lxing4812**
## Checklist

---
TODO: 

- [x]  Refactor the search component to use VarMenu and VarInput (p0)
- [x]  Increase the search for empty states (p0)
- [x] Optimize localSearch plugin data(p0)
- [x]  Keyboard Shortcuts (p1)

## Change information

---

### CODE REVIEW

**Comment by chouchouji** [comment]
I think adapting to Light-on-dark color scheme is necessary. 
image

**Review by haoziqaq — COMMENTED** [review]
A refactoring and readability optimization of the code is recommended. Because the functional logic of the markdown plug-in and search component is more complicated.

File: packages/varlet-ui/package.json [hunk 1]
```diff
@@ -58,16 +58,18 @@
     "@varlet/shared": "workspace:*",
     "@varlet/use": "workspace:*",
     "dayjs": "^1.10.4",
-    "decimal.js": "^10.2.1"
+    "decimal.js": "^10.2.1",
+    "minisearch": "^6.1.0",
+    "vue3-popper": "^1.5.0"
```
[the remark this answers is not in the document]
Comment (chouchouji) [reply packages/varlet-ui/package.json]: > Here should be move to devDependencies.

Isn't using vue3-popper components unreasonable? If there are some bugs in this component that affect our search function, I am not sure that the progress of their fix.

[the remark this answers is not in the document]
Comment (haoziqaq) [reply packages/varlet-ui/package.json]: Yes, we should use <var-menu> instead.

**Comment by chouchouji** [comment]
My bro, remember to add a todo list to plan tasks clearly.

**Pull request closed without merging**